be/src/util/buffered_stream.h

```diff
@@ -64,4 +72,41 @@ class BufferedInputStream {
     uint64_t _file_offset = 0;
 };
 
+class SharedBufferedInputStream : public IBufferedInputStream {
+public:
+    struct IORange {
+        int64_t offset;
+        int64_t size;
+        bool operator<(const IORange& x) const { return offset < x.offset; }
+    };
+    struct CoalesceOptions {
+        static constexpr int64_t MB = 1024 * 1024;
+        int64_t max_read_size = 16 * MB;
```

Comment:
Could you explain why not reuse `max_buffer_size`?

Reply:
they have different purpose:
1. max_buffer_size to separate small/large ranges
2. max_read_size to control max coalesced range.

But to why they dont share same value, I don't have much idea. It's the settings same to trino code.